### ISSUE TITLE
[FEATURE_REQUEST] static width and units for the network#speed module

### ISSUE THREAD

**Issue by ipuustin** [Alexays/Waybar#3851]
Currently the network#speed module has the bandwidth values length changing -- would it be possible to add padding to the values to prevent distracting resizing?

For example, go from `△ 4.0B/s  ▽ 128.0B/s` to `△   4.0B/s  ▽ 128.0B/s`

While at it, could we configure the lower and upper limit of the units used? I would like to have a kB (or MB) as the base unit to make the network traffic speed recognizable by a glance. So go from `△ 4.0B/s  ▽ 128.0B/s` to `△   0.0kB/s  ▽   0.1kB/s`.

**Comment by michaeltraxler** [comment]
Yes, I agree, that feature would be very helpful. It is quite irritating, that the text in the waybar moves left to right when the used bandwidth changes. It should be a static text format, only the numbers should change, not the position. Otherwise it distracts too much, as the whole text moves around. 
Or can you point to the position in the source code, where I can change the format of the output myself?

**Comment by theatischbein** [comment]
Isn't it already there?

I had the same problem and looked into it.

I change my config to this

```
    "network": {
       ...
        "format-wifi": "{essid:.15} {signalStrength}% {bandwidthTotalBytes:>3} ",
       ...
    },
```

The magic happens here `{bandwidthTotalBytes:>3}`.

With `{var:>3}` I tell the formater to reverse space for 3 digits in front or do a right align.

The code base is here: https://github.com/Alexays/Waybar/blob/master/include/util/format.hpp#L65C1-L79C6

**Comment by michaeltraxler** [comment]
Very strange. I don't get any difference, also with ":>20".
I have "Waybar v0.12.0" on my system.

**Comment by theatischbein** [comment]
here is my full network configuration:

```
    "network": {
        "format-wifi": "{essid} {signalStrength}% {bandwidthTotalBytes:>3} ",
//        "format-wifi": "{essid:.40} Down: {bandwidthDownBytes:>10} / Up: {bandwidthUpBytes:>10}",
        "format-ethernet": "{ipaddr}/{cidr} ",
        "tooltip-format": "{essid} on {ifname} ({frequency}) via {gwaddr} ",
        "format-linked": "{ifname} (No IP) ",
        "format-disconnected": "Disconnected ⚠",
        "format-alt": "{ifname}: {ipaddr}/{cidr}",
        "interval": 3,
    },
```

Maybe some other option does something strange?

**Comment by michaeltraxler** [comment]
No change with your settings. But I found a way to fix it. In the file `style.css`  I changed the
font-family from 
`font-family:   "Source Sans Pro", "FontAwesome 6 Free";`
to
` font-family:   "Hack", "FontAwesome 6 Free";`

Now the rates follow the ":>" option. But the length, e.g. ":>20" is completely ignored. But it is still fine for me now :-)

**Comment by layus** [comment]
The length is ignored by design, but feel free to propose changes and improvements. The code is in https://github.com/Alexays/Waybar/blob/master/include/util/format.hpp and the reason why numeric widths are ignored is explained there.

I feel like you had an issue with your fonts not having a fixed-size space. With fontconfig there was (is ?) a bug where spaces are taken from the most recently used font, so spaces after an icon are always taken from the icon font, and fontawesome does not have "fixed-width" spaces, and certainly not spaces matching the width of your fixed-width font.
That is why I needed this workaround:

https://github.com/layus/dotfiles/blob/ae20ac786222c64675ec11eeb300ca66370ed9ac/home/dotfiles/waybar/config#L139-L140

**Comment by PHPirates** [comment]
For the second part of the issue, namely fixing the units, you can use the script from https://github.com/Alexays/Waybar/issues/2162#issuecomment-1537366689 and change the `format_size` function for example to

```
def format_size(size):
    """ Always show speed in Mbit: number of bytes divided by (2^10 * 2^10 / 8)"""
    return f"{size / 2**17:3.0f} Mbit"
```

<img width="442" height="78" alt="Image" src="https://github.com/user-attachments/assets/75d34ecd-a7f9-4dd7-9a6f-17581b900dbf" />

<img width="442" height="70" alt="Image" src="https://github.com/user-attachments/assets/dd548699-22f6-4d20-bdcf-b3a6842a621e" />

**Comment by Alexays** [comment]
Consolidating this with #2162, which is the older canonical request for compact / fixed-width network speed formatting (including pinning the base unit to MB/s etc.) — let's keep the discussion in one place there. For the two asks here: right-alignment/static width is already available today via format specs, e.g. `{bandwidthTotalBytes:>10}` (as @theatischbein showed) — note the numeric *length* is intentionally ignored, and @layus explained the fontconfig "space taken from the icon font" gotcha that makes a fixed-width font matter. The configurable base-unit part is the genuinely-unimplemented FR, and it lives on in #2162. Closing as a duplicate 🙏 — a PR to `include/util/format.hpp` for unit pinning would be very welcome.
